NEXT-30083 - Documentation for Quotes (#1249)
shopware/docs: products/extensions/b2b-components/quotes-management/quotes-conversion.md

---
nav:
  title: Quotes conversion
  position: 30

---

# Quotes conversion

> We allow customers to convert their shopping carts into quotes to facilitate seamless processing. We introduce 2 new services to handle the conversion process

Reviewer: ```suggestion
Customers can convert their shopping carts into quotes to facilitate seamless processing. There are two new services to handle the conversion process :
```

> **Cart to quote converter:**

Reviewer: ```suggestion
## Cart to quote converter
```

> When a customer wants to request a quote from their shopping cart, we basically convert a cart to an order, then process to enrich the data for quote. the method `convertToQuote` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\CartToQuote\CartToQuoteConverter` is responsible for that process

Reviewer: ```suggestion
When a customer wants to request a quote for their shopping cart, the process involves converting a cart to an order and then proceeding to enrich the data for the quote. The method `convertToQuote` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\CartToQuote\CartToQuoteConverter` is responsible for this process.
```

```PHP
use Shopware\Core\Checkout\Cart\Order\OrderConversionContext;
use Shopware\Core\Checkout\Cart\Cart;
use Shopware\Core\Checkout\Cart\Order\OrderConverter;

public function convertToQuote(Cart $cart, SalesChannelContext $context, OrderConversionContext $orderContext = null): Quote
    {
        $order = $this->orderConverter->convertToOrder($cart, $context, $orderContext);
        
        $quote = $order;
        
        //enrich quote data
        
        //enrich quote line-items
        
        return $quote;
    }
```

> **Quote to cart converter:**

Reviewer: ```suggestion
## Quote to cart converter
```

> When a customer wants to place an order based on a quote, we will create a new cart based on quote data, the method `convertToCart` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\QuoteToCart\QuoteToCartConverter` is responsible for that process

Reviewer: ```suggestion
When a customer wants to place an order based on a quote, a new cart is created based on the quote data. The method `convertToCart` of class `Shopware\Commercial\B2B\QuoteManagement\Domain\QuoteToCart\QuoteToCartConverter` is responsible for this process.
```
